projector-plugin/src/main/kotlin/Utils.kt

```
@@ -25,10 +22,150 @@ import java.awt.Toolkit
          System.getProperty("jdk.attach.allowAttachSelf")?.toBoolean() == true
 }
 
+fun setSystemProperty(name: String, value: String?) {
+  if (value == null) {
+    System.clearProperty(name)
+  }
+  else {
+    System.setProperty(name, value)
+  }
+}
+
+private const val SUBSYSTEM = "PROJECTOR_SERVICE_CONFIG"
+const val PROJECTOR_RW_TOKEN_KEY = "PROJECTOR_RW_TOKEN"
+const val PROJECTOR_RO_TOKEN_KEY = "PROJECTOR_RO_TOKEN"
+
+private fun createCredentialAttributes(key: String) = CredentialAttributes(generateServiceName(SUBSYSTEM, key))
+
+fun loadToken(key: String): String? {
+  val credentialAttributes = createCredentialAttributes(key)
+  return PasswordSafe.instance.getPassword(credentialAttributes)
+}
+
+fun storeToken(key: String, value: String?) {
+  val attributes = createCredentialAttributes(key)
+  val credentials = Credentials(user = null, password = value)
+  PasswordSafe.instance.set(attributes, credentials)
+}
+
+fun migrateTokensToSecureStorage() {
+  copyTokenToSecureStorage(PROJECTOR_RW_TOKEN_KEY, "rwToken")
+  copyTokenToSecureStorage(PROJECTOR_RO_TOKEN_KEY, "roToken")
+  removeTokensFromXmlStorage()
+}
+
+private fun loadTokenFromXmlStorage(tokenName: String): String? {
+  val file = Paths.get(PathManager.getOptionsPath(), ProjectorConfig.STORAGE_NAME).toFile()
+  var result: String? = null
+
+  if (file.exists()) {
+    val factory = DocumentBuilderFactory.newInstance()
+
+    try {
+      val builder = factory.newDocumentBuilder()
+      val doc = builder.parse(file)
+      val elements = doc.getElementsByTagName("option")
+
+      for (i in 0 until elements.length) {
+        val e = elements.item(i)
+        val attrs = e.attributes
+        val name = attrs.getNamedItem("name")
+
+        if (name.nodeValue == tokenName) {
+          val token = attrs.getNamedItem("value")
+          result = token.nodeValue
+          break
+        }
+      }
+    }
+    catch (e: Exception) {
+
+    }
```

Is it OK to ignore the exception?

---

Fixed